source/deployment/sso-gitlab.rst

```diff
@@ -1,62 +1,143 @@
+
+Mattermost Enterprise Edition E20
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Mattermost Enterprise Edition E20 customers can set up a new GitLab OpenID Connect identity provider, or they can convert an existing OAuth 2.0 configuration for GitLab to the OpenID Connect standard. From Mattermost Enterprise Edition v5.33, OAuth 2.0 is being deprecated and replaced by OpenID Connect. Refer to product documentation to `convert your existing OAuth 2.0 service provider configuration for GitLab <https://docs.mattermost.com/deployment/converting-oauth20-service-providers-to-openidconnect.html>`__ to the OpenID Connect Standard. 
+
+1. Log in to Mattermost, then go to **System Console > Authentication > OpenID Connect**.
+2. Select **GitLab** as the service provider.
+3. Enter the **Gitlab Site URL** of your GitLab instance. If your GitLab instance is not set up to use SSL, start the URL with ``http://`` instead of ``https://``. If you're using GitLab itself as your provider, use ``gitlab.com``.
+4. The **Discovery Endpoint** for OpenID Connect with GitLab is prepopulated with ``https://gitlab.com/.well-known/openid-configuration``.
+5. Paste the **Application ID** from GitLab as the **Client ID** in Mattermost.
+6. Paste the **Application Secret Key** from GitLab as the **Client Secret** in Mattermost.
+7. Select **Save**.
+
+Alternatively, you can add the GitLab settings directly to the ``config.json`` file on your Mattermost server.
+
+1. Open ``config.json`` as *root* in a text editor. It’s usually in ``/opt/mattermost/config`` but it might be elsewhere on your system.
+2. Locate the ``GitLabSettings`` section, then add or update the following information:
+
+.. code-block:: text
+
+  "GitLabSettings": {
+    "Enable": true,
+    "Secret": "d4ff651332d26b0f76a18141ec6b72d59e04ad",
+    "Id": "f4c18eb49f437407758de8b9803668ae6",
+    "Scope": "profile openid email",
+    "AuthEndpoint": "",
+    "TokenEndpoint": "",
+    "UserApiEndpoint": "",
+    "DiscoveryEndpoint": "https://gitlab.com/.well-known/openid-configuration",
+    "ButtonText": "",
+    "ButtonColor": ""
+  }
+
+3. Save your changes, then restart your Mattermost server. After the server restarts, users must change their sign-in method before they can sign in with GitLab.
+
+(Optional) Step 3: Force users to sign up using SSO only
+--------------------------------------------------------
+
+To force all users to sign up with SSO only, go to **System Console > Authentication > Email**, then set **Enable sign-in with email** to **false**. 
+
+Alternatively, add this setting to the ``config.json`` file directly in the ``EmailSettings`` section and set ``EnableSignUpWithEmail`` to ``false``. You must save your changes and restart the Mattermost server. Users must change their sign-in method before they can sign in to Mattermost with GitLab.
```

Comment:
More a question on this for my understanding...  If users are forced to sign up with SSO only, and signing in with email is set to "false", are they still able to go to account settings to change their sign-in method from email to GitLab (or is this managed by the setting in the system console/config.json and automatically shows GitLab as sign-in method?)